Display.cpp

```diff
@@ -0,0 +1,87 @@
+#include "Display.h"
+#include <iostream>
+
+using namespace std;
+
+Display::Display(vector<LogLine>& set) : dataset(set) {}
+
+void Display::printPV(string pid, string vid)
+{
+    for(auto index : dataset)
+    {
+        if(index.getProcessId() == pid && index.getThreadId() == vid)
+        {
+            print(index);//write print final print function
+        }
+    }
+}
+void Display::printTimeRange(string start, string end)
+{
+    for(auto index : dataset)
+    {
+        doParsing(start);
+        map<string,int> uTimemap;
+        uTimemap = index.getTime();
+        if(uTimemap.count("Hour") && uTimemap.count("Minute") && uTimemap.count("Second") && uTimemap.count("Millisecond"))
+        {
+            if(uTimemap["Hour"]>=hour && uTimemap["Minute"]>=minute && uTimemap["Second"]>=second && uTimemap["Millisecond"]>=millisecond)
+            {
+                doParsing(end);
+            }
+            if(uTimemap["Hour"]<=hour && uTimemap["Minute"]<=minute && uTimemap["Second"]<=second && uTimemap["Millisecond"]<=millisecond)
+            {
+                print(index);//write print final print function
+            }
+        }
+    }
+}
+void Display::print(LogLine lobj)
+{
+    map<string,int> time = lobj.getTime();
+
+    cout<<lobj.getDate()<<" "<<time["Hour"]<<":"<<time["Minute"]<<":"<<time["Second"]<<"."<<time["Millisecond"]<<" "<<lobj.getProcessId()<<" "<<lobj.getThreadId()<<" "<<lobj.getLogLevel()<<" "<<lobj.getLogMessage()<<endl;
+}
+void Display::doParsing(string input)
+{
+    int set_count =0, startIndex =0;
+    string word;
+    for(int i=0;i<input.length();i++)
+    {
+        if(input[i] ==':'){
```

Comment:
How are you parsing millisecs using this logic? millisec wont end with ' : '

Reply:
done